haystack/tools/component_tool.py

```diff
@@ -0,0 +1,296 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
```

Comment:
`to_dict`/`from_dict` methods are missing. They are essential to enable proper YAML serde.
They should be similar to those shown in the POC PR: https://github.com/deepset-ai/haystack-experimental/pull/164